apps/dh/api-dh/source/DataHub.WebApi/Clients/Wholesale/ProcessManager/OrchestrationInstanceMapperExtensions.cs

```diff
@@ -0,0 +1,185 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Energinet.DataHub.ProcessManager.Api.Model;
+using Energinet.DataHub.ProcessManager.Api.Model.OrchestrationInstance;
+using Energinet.DataHub.ProcessManager.Orchestrations.Processes.BRS_023_027.V1.Model;
+using Energinet.DataHub.WebApi.Clients.Wholesale.v3;
+
+namespace Energinet.DataHub.WebApi.Clients.Wholesale.ProcessManager;
+
+/// <summary>
+/// Extension methods to help us map to and from "old" and "new" Calculation types.
+/// </summary>
+public static class OrchestrationInstanceMapperExtensions
+{
+    /// <summary>
+    /// Map from "old" Calculation types to "new" types.
+    /// </summary>
+    public static (OrchestrationInstanceLifecycleStates? LifecycleState, OrchestrationInstanceTerminationStates? TerminationState)
+        MapToLifecycleState(this CalculationOrchestrationState state)
+    {
+        switch (state)
+        {
+            case CalculationOrchestrationState.Scheduled:
+                return (OrchestrationInstanceLifecycleStates.Pending, null);
+
+            case CalculationOrchestrationState.Started:
+            case CalculationOrchestrationState.Calculating:
+            case CalculationOrchestrationState.Calculated:
+            case CalculationOrchestrationState.ActorMessagesEnqueuing:
+            case CalculationOrchestrationState.ActorMessagesEnqueued:
+                return (OrchestrationInstanceLifecycleStates.Running, null);
+
+            case CalculationOrchestrationState.CalculationFailed:
+            case CalculationOrchestrationState.ActorMessagesEnqueuingFailed:
+                return (OrchestrationInstanceLifecycleStates.Terminated, OrchestrationInstanceTerminationStates.Failed);
+
+            case CalculationOrchestrationState.Completed:
+                return (OrchestrationInstanceLifecycleStates.Terminated, OrchestrationInstanceTerminationStates.Succeeded);
+
+            case CalculationOrchestrationState.Canceled:
+                return (OrchestrationInstanceLifecycleStates.Terminated, OrchestrationInstanceTerminationStates.UserCanceled);
+
+            default:
+                throw new InvalidOperationException($"Invalid {nameof(CalculationOrchestrationState)} '{state}'; cannot be mapped.");
+        }
+    }
+
+    /// <summary>
+    /// Map from "old" Calculation types to "new" types.
+    /// </summary>
+    public static CalculationTypes MapToCalculationType(
+        this v3.CalculationType calculationType)
+    {
+        return Enum
+            .TryParse<CalculationTypes>(
+                calculationType.ToString(),
+                ignoreCase: true,
+                out var calculationTypeResult)
+            ? calculationTypeResult
+            : throw new InvalidOperationException($"Invalid CalculationType '{calculationType}'; cannot be mapped.");
```

Comment:
suggestion: i know this is temporary code, but is it possible to use an explicit mapping here so that we can benefit from the compilers exhaustiveness checks?